tests/test_decimal.py

```diff
@@ -0,0 +1,115 @@
+        (123.4567, 2, "123.4567"),
+        (123.4500, 2, "123.45"),
+        (0.0, 2, "0.00"),
+        (0.1, 2, "0.10"),
+        (0.01, 2, "0.01"),
+        (0.001, 2, "0.001"),
+        (1000000.0, 2, "1000000.00"),
+        (1000000.10, 2, "1000000.10"),
+        (-123.45, 2, "-123.45"),
+        (-123.4500, 2, "-123.45"),
+        (-123.4567, 2, "-123.4567"),
+        (123.0, 3, "123.000"),
+        (123.4, 3, "123.400"),
+        (123.456789, 4, "123.456789"),
+    ],
+)
+def test_format_decimal_places_float(number, decimal_places, expected):
+    result = format_decimal_places(number, decimal_places=decimal_places)
+
+    assert isinstance(result, str)
+    assert result == expected
+    assert float(result) == pytest.approx(number)
+
+
+@pytest.mark.parametrize(
+    "number,decimal_places,expected",
+    [
+        ("123", 2, "123.00"),
+        ("123.4", 2, "123.40"),
+        ("123.40", 2, "123.40"),
+        ("123.4567", 2, "123.4567"),
+        ("123.4500", 2, "123.45"),
+        ("0", 2, "0.00"),
+        ("0.1", 2, "0.10"),
+        ("0.01", 2, "0.01"),
+        ("0.001", 2, "0.001"),
+        ("1000000", 2, "1000000.00"),
+        ("1000000.10", 2, "1000000.10"),
+        ("-123.45", 2, "-123.45"),
+        ("-123.4500", 2, "-123.45"),
+        ("-123.4567", 2, "-123.4567"),
+        ("123", 3, "123.000"),
+        ("123.4", 3, "123.400"),
+        ("123.456789", 4, "123.456789"),
+    ],
+)
+def test_format_decimal_places_str(number, decimal_places, expected):
+    result = format_decimal_places(number, decimal_places=decimal_places)
+
+    assert isinstance(result, str)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "number,decimal_places,expected",
+    [
+        (Decimal("123"), 2, Decimal("123.00")),
+        (Decimal("123.4"), 2, Decimal("123.40")),
+        (Decimal("123.40"), 2, Decimal("123.40")),
+        (Decimal("123.4567"), 2, Decimal("123.4567")),
+        (Decimal("123.4500"), 2, Decimal("123.45")),
+        (Decimal("0"), 2, Decimal("0.00")),
+        (Decimal("0.1"), 2, Decimal("0.10")),
+        (Decimal("0.01"), 2, Decimal("0.01")),
+        (Decimal("0.001"), 2, Decimal("0.001")),
+        (Decimal("1000000"), 2, Decimal("1000000.00")),
+        (Decimal("1000000.10"), 2, Decimal("1000000.10")),
+        (Decimal("-123.45"), 2, Decimal("-123.45")),
+        (Decimal("-123.4500"), 2, Decimal("-123.45")),
+        (Decimal("-123.4567"), 2, Decimal("-123.4567")),
+        (Decimal("123"), 3, Decimal("123.000")),
+        (Decimal("123.4"), 3, Decimal("123.400")),
+        (Decimal("123.456789"), 4, Decimal("123.456789")),
+    ],
+)
+def test_format_decimal_places_decimal(number, decimal_places, expected):
+    result = format_decimal_places(number, decimal_places=decimal_places)
+
+    assert isinstance(result, Decimal)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "decimal_places, expected",
+    [
+        (0, "123.456789"),
+        (2, "123.456789"),
```

Comment:
This throws me.

Reply:
Ah, yeah.. Not in love with the name of the function, which may go a little ways to explain what the purpose is (beyond, you know, some actual docs....).

For display in a template, I needed a quick way to format a dollar amount to two decimal places without rounding.

E.g.

123.4 -> 123.40
123.45 -> 123.45
123.456 -> 123.456
123.4567 -> 123.4567
123.45670 -> 123.4567

and so on.

Reply:
Or three decimal places, which is why I added the `decimal_places` arg. We have some tracts of land that have fractional acres, but the majority are whole numbers. So I wanted to just show 123 if the acres were 123.000, but 123.456 if the acres were 123.456.

I don't think I've actually pushed that local change actually.

Reply:
I need to clean it up a bit to clarify things, because even I'm confusing myself thinking through it right now.. 😅